samples/AdvancedPushNotifications/src/main/java/com/swrve/sdk/demo/BigNotificationSwrveGcmIntentService.java

```diff
@@ -0,0 +1,95 @@
+package com.swrve.sdk.demo;
+
+import android.app.Notification;
+import android.app.PendingIntent;
+import android.content.ContentResolver;
+import android.graphics.Bitmap;
+import android.graphics.BitmapFactory;
+import android.media.RingtoneManager;
+import android.net.Uri;
+import android.os.Bundle;
+import android.support.v4.app.NotificationCompat;
+
+import com.squareup.picasso.Picasso;
+import com.swrve.sdk.SwrveHelper;
+import com.swrve.sdk.gcm.SwrveGcmIntentService;
+
+import java.io.IOException;
+
+/*
+ * Use this class when you want to create richer local notifications form a push notification
+ * using the BigPicture style and grouping and/or others.
+ * http://developer.android.com/guide/topics/ui/notifiers/notifications.html
+ * http://developer.android.com/training/notify-user/expanded.html
+ * http://developer.android.com/training/wearables/notifications/stacks.html
+ */
+public class BigNotificationSwrveGcmIntentService extends SwrveGcmIntentService {
+
+    @Override
+    public Notification createNotification(Bundle msg, PendingIntent contentIntent) {
+        String msgText = msg.getString("text");
+        if (!SwrveHelper.isNullOrEmpty(msgText)) {
+            // Customizable title with the custom payload 'title'
+            String msgTitle = msg.getString("title");
+            if (SwrveHelper.isNullOrEmpty(msgTitle)) {
+                msgTitle = "Send 'title' payload to customize";
+            }
+            // Customizable big style summary with the custom payload 'summary'
+            String msgSummary = msg.getString("summary");
+            if (SwrveHelper.isNullOrEmpty(msgSummary)) {
+                msgSummary = "Send 'summary' payload to customize";
+            }
+            // Group notifications
+            String msgGroup = msg.getString("group");
+            if (SwrveHelper.isNullOrEmpty("group")) {
+                msgGroup = "group_key_emails";
+            }
+
+            // Set the large icon downloaded from the custom payload 'image_url' or a default
+            Bitmap largeBitmap = BitmapFactory.decodeResource(getResources(), R.drawable.default_large_image);
+            String imageUrl = msg.getString("image_url");
+            if (!SwrveHelper.isNullOrEmpty(imageUrl)) {
+                try {
+                    largeBitmap = Picasso.with(this).load(imageUrl).get();
```

Comment:
We are not pulling it into our SDK, just the samples. And a customer use case is specifically downloading an image from a custom payload URL.

The more in depth examples we give the less work they and CSM have to do.